### ISSUE TITLE
Document preview for object fields with name "title" crash on reload

### ISSUE THREAD

**Issue by jfulse** [sanity-io/sanity#3010]
**Describe the bug**

We have some object fields (for localization) used in the studio list previews (specyfing what key to use in `preview.prepare`). This hasn't been a problem until now, but in v2.23.1, the Studio document view crashes on reload, apparently in the [ObserveForPreview](https://github.com/sanity-io/sanity/blob/31133c55b02783d3472d7d0237ac19de2bb7aa67/packages/@sanity/base/src/preview/components/ObserveForPreview.tsx) component.

It appears there is only an issue if the field name is `title`.

**To Reproduce**

1. Use a schema similar to:
```
const schema = {
  name: "test
  title: "Test
  type: "document",
  fields: [{
    name: "title",
    type: "object",
    title: "Localized test field",
    initialValue: { key: "value" },
  }],
  preview: {
    select: { titleObject: "title" },
    prepare: ({ titleObject }) => ({ title: titleObject.key }),
  }
}
```
2. Navigate to the document editor in the Studio
3. Reload the page
4. The page crashes as react tries to render an object

**Expected behavior**

The Studio doesn't crash

**Screenshots**
![Skjermbilde 2021-12-15 kl  13 37 08](https://user-images.githubusercontent.com/13033324/146187863-fabd5b40-70cc-454c-b6c2-0679b48bd980.png)

**Which versions of Sanity are you using?**

@sanity/cli             2.19.1 (latest: 2.23.1)
@sanity/base            2.23.1 (up to date)
@sanity/cli             2.23.1 (up to date)
@sanity/core            2.23.1 (up to date)
@sanity/default-layout  2.23.1 (up to date)
@sanity/default-login   2.23.1 (up to date)
@sanity/desk-tool       2.23.1 (up to date)
@sanity/imagetool       2.22.3 (up to date)
@sanity/react-hooks     2.23.1 (up to date)
@sanity/vision          2.23.1 (up to date)

**What operating system are you using?**

macOS Big Sur

**Which versions of Node.js / npm are you running?**

8.1.3
v16.11.0

**Comment by bjoerge** [comment]
Hi @jfulse, and thanks for reporting. I'm not able to reproduce the issue you're having with the given example schema. What fields does your `title` type have here? Could you verify that you don't have any documents where the type of `title.key` is an object?

**Comment by KevinBodi** [comment]
@bjoerge We have this exact issue here and I think I came closer to instructions on how to reproduce it https://github.com/sanity-io/sanity/issues/3004

**Comment by jfulse** [comment]
> I'm not able to reproduce the issue you're having with the given example schema.

Sorry about that! From @KevinBodi's issue it looks like it might only happen with an image in the preview?

> Could you verify that you don't have any documents where the type of title.key is an object?

Pretty sure we don't! Also the crash doesn't happen when navigating to the document from the root view, only when reloading on the document.

**Comment by bjoerge** [comment]
Thanks all for providing steps to reproduce! We're rolling out a fix for this asap.

**Comment by bjoerge** [comment]
Fixed in [v2.23.2](https://github.com/sanity-io/sanity/releases/tag/v2.23.2)

**Comment by jfulse** [comment]
Great, thanks!
